tests/test_list.py

```diff
@@ -0,0 +1,130 @@
+from typing import Any
+
+from .mockers import (GitRepositorySandbox, assert_command, launch_command, mock_run_cmd)
+
+
+class TestList:
+
+    def setup_method(self) -> None:
+        self.repo_sandbox = GitRepositorySandbox()
+
+        (
+            self.repo_sandbox
+            # Create the remote and sandbox repos, chdir into sandbox repo
+            .new_repo(self.repo_sandbox.remote_path, "--bare")
+            .new_repo(self.repo_sandbox.local_path)
+            .execute(f"git remote add origin {self.repo_sandbox.remote_path}")
+            .execute('git config user.email "tester@test.com"')
+            .execute('git config user.name "Tester Test"')
+        )
+
+    def test_list(self, mocker: Any) -> None:
+        """
+        Verify behaviour of a 'git machete list' command.
+        """
+        mocker.patch('git_machete.utils.run_cmd', mock_run_cmd)  # to hide git outputs in tests
+
+        (
+            self.repo_sandbox.new_branch("master")
+                .commit("master commit.")
+                .new_branch("develop")
+                .commit("develop commit.")
+                .new_branch("feature_0")
+                .commit("feature_0 commit.")
+                .new_branch("feature_0_0")
+                .commit("feature_0_0 commit.")
+                .new_branch("feature_0_0_0")
+                .commit("feature_0_0_0 commit.")
+                .check_out("feature_0")
+                .new_branch("feature_0_1")
+                .commit("feature_0_1 commit.")
+                .check_out("develop")
+                .new_branch("feature_1")
+                .commit("feature_1 commit.")
+        )
+        launch_command("discover", "-y")
```

Comment:
Hmmm maybe better just fill up .git/machete file explicitly? It'll make it more clear how the branch layout is supposed to look

Reply:
Hmm, i can just add the expected layout in the comment below, wdyt?

Reply:
like: 
```
        # content of the machete definition file:
        """
        master
        develop
            feature_0
                feature_0_0
                    feature_0_0_0
                feature_0_1
            feature_1
        """
```

Reply:
Could be... but it might get outdated with the changes to the test. TBH I'd just overwrite machete file explicitly, no need to kinda-test `discover` in an otherwise non-`discover`-related test

Reply:
@PawelLipski I tried overwriting def file but i still had to leave the creation of the branches and the commits so in the end it looked less clear plus it could get outdated as well, i don't see why it would be better

Reply:
Hmm, my bad, it would not get updated as the tests will probably fail, will try again